src/GafferScene/Instancer.cpp

```diff
@@ -857,10 +1471,99 @@ void Instancer::compute( Gaffer::ValuePlug *output, const Gaffer::Context *conte
 		static_cast<AtomicCompoundDataPlug *>( output )->setValue( result );
 		return;
 	}
+	else if( output == variationsPlug() )
+	{
+		// Compute the number of variations by accumulating massive lists of unique hashes from all EngineDatas
+		// and then counting the total number of uniques
+		tbb::spin_mutex locationMutex;
+		std::vector< std::unique_ptr< EngineData::PrototypeHashes > > perLocationHashes;
+
+		std::function<bool(const GafferScene::ScenePlug *, const GafferScene::ScenePlug::ScenePath &)> functor =[this, &locationMutex, &perLocationHashes]( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
```

Comment:
As above, I think we can use `auto` here and avoid the `std::function`.

Reply:
Yep.